### PR TITLE
feat(python): Prettify show_versions

### PR DESCRIPTION
**Pull request by mcrumiller**
Implements #8625. New look:

```
>>> pl.show_versions()
--------Version info---------
Polars:      0.17.11
Index type:  UInt32
Platform:    Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python:      3.11.3 (main, Apr 15 2023, 14:44:51) [GCC 11.3.0]

----Optional dependencies----
numpy:       1.24.2
pandas:      2.0.0
pyarrow:     11.0.0
connectorx:  <not installed>
deltalake:   0.8.1
fsspec:      2023.4.0
matplotlib:  3.7.1
xlsx2csv:    0.8.1
xlsxwriter:  3.1.0
```

### CODE REVIEW

File: py-polars/polars/utils/show_versions.py [hunk 1]
```diff
@@ -13,36 +13,49 @@ def show_versions() -> None:
...
     deps = _get_dependency_info()
+
+    # determine key length for alignment
+    keylen = (
+        max(
+            [
+                len(x)
+                for x in [*deps.keys(), "Polars", "Index type", "Platform", "Python"]
+            ]
+        )
```
Comment (stinodego) [inline py-polars/polars/utils/show_versions.py:46]: You can leave out the brackets here, very slight performance optimization.
```suggestion
        max(
            len(x)
            for x in [*deps.keys(), "Polars", "Index type", "Platform", "Python"]
        )
```

**Comment by stinodego** [comment]
Looks great! One very small detail I'd like to see changed, then we can merge.

File: py-polars/polars/utils/show_versions.py (hunk 1 above)
Comment (stinodego) [reply py-polars/polars/utils/show_versions.py]: Gotta run the autoformatter too!

**Comment by mcrumiller** [comment]
Thanks @stinodego. The auto-formatter tends to go nuts with indendation so I'm happy to say goodbye to any unnecessary brackets.

Edit: looks like committing on the spot didn't do it, let me rerun the formatter.

**Pull request merged**